websurvey/src/App.svelte

```diff
@@ -0,0 +1,46 @@
+<script>
+	import { Router, Route, Link } from "svelte-routing";
+	import Home from "./routes/Home.svelte";
+	import Form from "./routes/Form.svelte";
+
+	export let url = "";
+
+	import typeformData from "./typeformData.js";
+
+	const { fields } = typeformData;
+
+	let ref;
+	let currentIndex = 0;
+
+	const setInitialRef = (index) => {
+		ref = fields[index].ref;
+		console.log(
+			`The starting index is ${index} and the starting ref is ${ref}`
+		);
+	};
+
+	const setCurrentRef = (index) => {
+		ref = fields[index].ref;
+	};
+
+	const indexUpdate = (e) => {
+		currentIndex = e.detail;
+		setCurrentRef(currentIndex);
+
+		console.log(
+			`The current index is ${currentIndex} and the current ref is ${ref}`
+		);
+	};
+</script>
+
+<main>
+	<Router {url}>
+		<nav>
+			<Link to="/">Home</Link>
+		</nav>
+		<Route path="/">
+			<Home {setInitialRef} />
+		</Route>
+	</Router>
+	<Form on:indexUpdate={indexUpdate} {currentIndex} {ref} />
```

Comment:
It's a clever way to get it all to update! But you've essentially invented your own router ;)

Can you use the :params functionality of `svelte-routing` to get the ref from the url to your component? i.e. `/:ref`? 

This should give you a hint: https://github.com/EmilTholin/svelte-routing/blob/master/src/Route.svelte#L42

Reply:
Focus on that: just getting the router library to pass you the ref param to the component. Then you can get the component to "react" to a change in that param and switch up the field it shows.

Reply:
I get you – I logged out the `params` object and it was interesting. Basically what we're doing is giving it a key and a value based on whatever we name the prop and the pathname.

What still doesn't make sense to me though is the `to` prop on the `Link` component. To me this is just a hardcoded string that says `/:ref` and has no "reactivity" to changes in the `path` prop that's passed to the `Router`. 

I feel like this should be a variable which is how I've ended up in a loop of passing `ref` from parent to child and back up to parent again – unless I'm missing something?

Reply:
Heh, yes, you should avoid that loop!

`ref` should flow from parent to chlid. Really, from Route to component. If you look at that code, we know that any `routeParams` will be passed as parameters to the component, no?

Let me know if you want to go through this in person on Monday

Reply:
Yes, agreed. I've cleaned up the code a bit in `App.svelte` to make it clearer which way the data is flowing i.e. from `Route` to component. 

We now have event dispatchers for two child components `Home` and `Form` which "listen" for a submit event and assign a value to the `ref` using the `index`.